setup.py

```diff
@@ -358,6 +358,7 @@ def write_version(filename: str = str(AIRFLOW_SOURCES_ROOT / "airflow" / "git_ve
     "bcrypt>=2.0.0",
     "flask-bcrypt>=0.7.1",
 ]
+polars = ["polars>=0.19.5"]
```

Comment:
I dont think this is right(?)
Polars should be extra dependency of common.sql provider (it should be defined in the provider.yaml)

I know we have pandas here but I think its due to legacy reasons (pandas used to be part of Airflow core)

Reply:
You are probably correct, @eladkal ; I mostly just mirrored what I saw `pandas` doing.

As for `common.sql`, [you mentioned about it in the issue as well](https://github.com/apache/airflow/issues/33911#issuecomment-1698925741), but I didn't understand what you meant there. Now that you say `provider.yaml`, I suppose I get it, you mean like [here](https://github.com/apache/airflow/blob/23e2c9527c1167b2262968cdad4d984708f4b2aa/airflow/providers/google/provider.yaml#L130)?

Reply:
Not quite... by having it as dependency of the provider you force every user of the provider to install this lib.
What we should do is have it as optional extra for the provider. Thus you need to define it in:
https://github.com/apache/airflow/blob/23e2c9527c1167b2262968cdad4d984708f4b2aa/airflow/providers/google/provider.yaml#L1191

then users can have it when they explicitly ask to install the extra :
`pip install apache-airflow-providers-google[polars]`

Reply:
Hm okay. I've removed `polars` from core dependencies (I think), and added it to `airflow/airflow/providers/google/provider.yaml`.

Sorry, I'm still not clear on which you file you mean by "the common.sql provider", what's the precise path to which you're referring?

Reply:
@eladkal can you please advise here?